### PR TITLE
Python: Add example of top-level module shadowing stdlib (v2)

### PR DESCRIPTION
**Pull request by RasmusWL**
This PR adds a test-case to show a specific kind of bad module resolution that happens when using `codeql database create`, since a locally defined module doesn't take precedence in module resolution. It turned out that we don't invoke the extractor in the same way when running our qltests, and that the way qltests behaves is more correct (compared with Python interpreter). Obvious fixes didn't work:

- Make extractor better: Hard, since you need to figure out exactly at what paths the Python interpreter is going to be invoked, so you can simulate adding that path as `sys.path[0]`.
- Change qltest extractor to match `codeql database create`: I naively attempted this, but it turned out to be a much bigger tasks than I am ready to take on right now.

So instead I changed the test manually to simulate what the result of `codeql database create` is (and even that was tricky). Reboot of https://github.com/github/codeql/pull/4655, but a bit more clear now that I've figured out what to do.

---

Kind of a bonus test hiding within all of this is the `unique_name`, which just shows that we're not able to resolve it properly when it resides in non-package directory... but it's not the essential part I'm trying to tests here, I think I just included it as a safety check.

### CODE REVIEW

**Comment by RasmusWL** [comment]
I accidentally included a change to a file that wasn't supposed to be part of this branch, and force pushed to remove this...

File: python/ql/test/library-tests/PointsTo/regressions/wrong/module-imports/conflict-stdlib/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+This test shows how we handle modules the shadow a module in the standard library.
```
Comment (tausbn) [inline python/ql/test/library-tests/PointsTo/regressions/wrong/module-imports/conflict-stdlib/README.md:1]:
```suggestion
This test shows how we handle modules that shadow a module in the standard library.
```

File: python/ql/test/library-tests/PointsTo/regressions/wrong/module-imports/conflict-stdlib/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+We manually replicate the behavior of `codeql database create --source-root <src-dir>`, which will use `-R <src-dir>`. By default, the way qltest invokes the extractor will cause different behavior. Therefore, we also need to move our code outside of the top-level folder, and it lives in `code-invalid-package-name/` -- notice that if we use `code` as the folder name, the extractor will treat it as if there is a package called `code` (note, `codeql database create` would not the folder `code` as a package when `code` is used as the `--source-root`).
```
Comment (tausbn) [inline python/ql/test/library-tests/PointsTo/regressions/wrong/module-imports/conflict-stdlib/README.md:3]:
```suggestion
We manually replicate the behavior of `codeql database create --source-root <src-dir>`, which will use `-R <src-dir>`. By default, the way qltest invokes the extractor will cause different behavior. Therefore, we also need to move our code outside of the top-level folder, and it lives in `code-invalid-package-name/` -- notice that if we use `code` as the folder name, the extractor will treat it as if there is a package called `code` (note, `codeql database create` would not treat the folder `code` as a package when `code` is used as the `--source-root`).
```
I inserted a missing verb, but I don't know if it was the intended verb.

**Review by tausbn — CHANGES_REQUESTED** [review]
Nice and clear example of the bad behaviour. :+1:

I have made two suggestion with regard to typos, otherwise this looks good to me.

**Review by tausbn — APPROVED** [review]

**Pull request merged**